src/engraving/rendering/dev/tlayout.cpp

```diff
@@ -1671,54 +1671,57 @@ void TLayout::layout(Expression* item, LayoutContext& ctx)
     }
 }
 
-void TLayout::layout(Fermata* item, LayoutContext& ctx)
+static void layoutFermata(const Fermata* item, const LayoutContext& ctx, Fermata::LayoutData* data)
 {
     const StaffType* stType = item->staffType();
 
-    item->setSkipDraw(false);
     if (stType && stType->isHiddenElementOnTab(ctx.conf().style(), Sid::fermataShowTabCommon, Sid::fermataShowTabSimple)) {
-        item->setSkipDraw(true);
+        data->isSkipDraw = true;
         return;
     }
+    data->isSkipDraw = false;
 
     Segment* s = item->segment();
-    item->setPos(PointF());
-    if (!s) {            // for use in palette
-        item->setOffset(0.0, 0.0);
-        RectF b(item->symBbox(item->symId()));
-        item->setbbox(b.translated(-0.5 * b.width(), 0.0));
-        return;
-    }
+    data->pos = PointF();
 
     if (item->isStyled(Pid::OFFSET)) {
-        item->setOffset(item->propertyDefault(Pid::OFFSET).value<PointF>());
+        const_cast<Fermata*>(item)->setOffset(item->propertyDefault(Pid::OFFSET).value<PointF>());
     }
+
     EngravingItem* e = s->element(item->track());
     if (e) {
         if (e->isChord()) {
             Chord* chord = toChord(e);
             Note* note = chord->up() ? chord->downNote() : chord->upNote();
             double offset = chord->xpos() + note->xpos() + note->headWidth() / 2;
-            item->movePosX(offset);
+            data->movePosX(offset);
         } else {
-            item->movePosX(e->x() - e->shape().left() + e->width() * item->staff()->staffMag(Fraction(0, 1)) * .5);
+            data->movePosX(e->x() - e->shape().left() + e->width() * item->staff()->staffMag(Fraction(0, 1)) * .5);
         }
     }
 
     String name = String::fromAscii(SymNames::nameForSymId(item->symId()).ascii());
     if (item->placeAbove()) {
         if (name.endsWith(u"Below")) {
-            item->setSymId(SymNames::symIdByName(name.left(name.size() - 5) + u"Above"));
+            //! NOTE It is not clear whether SymId is layout data or given data.
```

Comment:
The answer is: the symbol to be used _should_ be given data (or you could say: the _fermata type_ should given data and the symbol is a direct consequence of that), but the actual SymId is layout data, because each symbol has both an "above" and "below" version. This ugly piece of code tries to choose the correct version, while keeping the essentially same symbol.